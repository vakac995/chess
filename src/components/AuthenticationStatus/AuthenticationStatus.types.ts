import type { Optional } from '@/types';

export interface AuthenticationStatusProps {
  readonly compact?: Optional<boolean>;
}
