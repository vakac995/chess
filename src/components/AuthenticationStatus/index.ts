export { AuthenticationStatus } from './AuthenticationStatus';
export type { AuthenticationStatusProps } from './AuthenticationStatus.types';
