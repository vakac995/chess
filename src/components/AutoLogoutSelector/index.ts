export { AutoLogoutSelector } from './AutoLogoutSelector';
export type {
  AutoLogoutSelectorProps,
  AutoLogoutPreference,
  AutoLogoutMode,
  AutoLogoutModeOption,
} from './AutoLogoutSelector.types';
