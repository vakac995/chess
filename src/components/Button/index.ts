export { Button } from './Button';
export type { ButtonProps, ButtonIntent, ButtonSize } from './Button.types';
