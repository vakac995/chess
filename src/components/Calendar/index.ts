// Calendar components
export { Calendar, CalendarWrapper } from './Calendar';
export { SingleCalendar } from './SingleCalendar';
export { RangeCalendar } from './RangeCalendar';
export { DualCalendar } from './DualCalendar';

// Types
export type {
  CalendarMode,
  CalendarProps,
  CalendarWrapperProps,
  SingleCalendarProps,
  RangeCalendarProps,
  DualCalendarProps,
  BaseCalendarProps,
  CalendarState,
  CalendarSelection,
  CalendarContextValue,
  CalendarUtils,
  CalendarValidationResult,
  CalendarTheme,
  CalendarConfig,
} from './Calendar.types';

// Schemas
export {
  calendarModeSchema,
  dateSchema,
  optionalDateSchema,
  dateStringSchema,
  dateRangeSchema,
  calendarSelectionSchema,
  calendarStateSchema,
  calendarConfigSchema,
  calendarValidationUtils,
} from './Calendar.schemas';
