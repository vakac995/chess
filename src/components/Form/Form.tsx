import { FieldValues, FormProvider } from 'react-hook-form';
import type { FormProps } from './Form.types';

export const Form = <TFieldValues extends FieldValues>({
  form,
  onSubmit,
  children,
  className,
}: Readonly<FormProps<TFieldValues>>) => {
  return (
    <FormProvider {...form}>
      <form onSubmit={form.handleSubmit(onSubmit)} className={className}>
        {children}
      </form>
    </FormProvider>
  );
};
