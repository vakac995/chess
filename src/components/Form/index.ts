export { Form } from './Form';
export { FormField } from './FormField';
export { ErrorInfo } from './ErrorInfo';

export type { FormProps } from './Form.types';
export type { FormFieldProps, RenderPropInput } from './FormField.types';
