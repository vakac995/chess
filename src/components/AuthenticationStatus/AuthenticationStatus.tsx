import { useAuth } from '../../hooks/useAuth';
import { ErrorInfo } from '../../components/Form';

export function AuthenticationStatus() {
  const { user, isAuthenticated, isLoading, error, login, logout } = useAuth();

  const handleLogin = async () => {
    await login({ email: 'user@example.com', password: 'password123' });
  };

  const handleLogout = () => {
    logout();
  };

  return (
    <div className="p-4 my-4 bg-gray-100 border rounded shadow">
      <h2 className="text-xl font-semibold mb-3">Authentication Status (Debug)</h2>
      <p className="mb-2">
        Status:{' '}
        <span className={`font-medium ${isAuthenticated ? 'text-green-600' : 'text-red-600'}`}>
          {isAuthenticated ? 'Authenticated' : 'Not Authenticated'}
        </span>
      </p>

      {isLoading && <p className="text-blue-500 animate-pulse">Loading...</p>}

      {error && (
        <div className="my-2 p-2 bg-red-100 border border-red-300 rounded">
          <ErrorInfo error={error} />
        </div>
      )}

      {user && (
        <div className="mb-2 text-sm text-gray-700">
          <p>User ID: {user.id}</p>
          <p>Email: {user.email}</p>
          {user.username && <p>Username: {user.username}</p>}
        </div>
      )}

      <div className="flex gap-2 mt-4">
        <button
          onClick={handleLogin}
          disabled={isLoading || isAuthenticated}
          className="bg-blue-500 hover:bg-blue-600 text-white px-4 py-2 rounded disabled:opacity-50 disabled:cursor-not-allowed transition duration-150"
        >
          Log In (Test)
        </button>
        <button
          onClick={handleLogout}
          disabled={!isAuthenticated || isLoading}
          className="bg-gray-500 hover:bg-gray-600 text-white px-4 py-2 rounded disabled:opacity-50 disabled:cursor-not-allowed transition duration-150"
        >
          Log Out
        </button>
      </div>
    </div>
  );
}
