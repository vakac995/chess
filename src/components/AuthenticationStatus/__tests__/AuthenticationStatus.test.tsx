import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import type { Mock } from 'vitest'; // Import Mock type
import { AuthenticationStatus } from '../AuthenticationStatus';
import { useAuth } from '@/hooks';
import { ErrorInfo } from '@/components/Form';

// Mock the useAuth hook
vi.mock('@/hooks', () => ({
  useAuth: vi.fn(),
}));

// Mock the ErrorInfo component
vi.mock('@/components/Form', () => ({
  ErrorInfo: vi.fn(({ error }) => <div data-testid="error-info">{error?.message ?? 'Error'}</div>), // Changed || to ??
}));

describe('AuthenticationStatus', () => {
  let mockLogin: Mock;
  let mockLogout: Mock;

  beforeEach(() => {
    vi.clearAllMocks();
    mockLogin = vi.fn().mockResolvedValue(undefined);
    mockLogout = vi.fn();
  });

  const setup = (props: Partial<ReturnType<typeof useAuth>> & { compact?: boolean } = {}) => {
    const defaultAuthValues = {
      user: null,
      isAuthenticated: false,
      isLoading: false,
      error: null,
      login: mockLogin,
      logout: mockLogout,
    };
    (useAuth as Mock).mockReturnValue({ ...defaultAuthValues, ...props });
    return render(<AuthenticationStatus compact={props.compact} />);
  };

  it('renders not authenticated state by default', () => {
    setup();
    expect(screen.getByText(/Status:/i)).toHaveTextContent('Not Authenticated');
    expect(screen.getByRole('button', { name: /Log In \(Test\)/i })).toBeEnabled();
    expect(screen.getByRole('button', { name: /Log Out/i })).toBeDisabled();
  });

  it('renders authenticated state', () => {
    setup({ isAuthenticated: true, user: { id: '1', email: 'test@example.com' } });
    expect(screen.getByText(/Status:/i)).toHaveTextContent('Authenticated');
    expect(screen.getByText(/User ID: 1/i)).toBeInTheDocument();
    expect(screen.getByText(/Email: test@example.com/i)).toBeInTheDocument();
    expect(screen.getByRole('button', { name: /Log In \(Test\)/i })).toBeDisabled();
    expect(screen.getByRole('button', { name: /Log Out/i })).toBeEnabled();
  });

  it('calls login on login button click', async () => {
    setup();
    fireEvent.click(screen.getByRole('button', { name: /Log In \(Test\)/i }));
    await waitFor(() => expect(mockLogin).toHaveBeenCalledTimes(1));
    expect(mockLogin).toHaveBeenCalledWith({ email: 'user@example.com', password: 'password123' });
  });

  it('calls logout on logout button click', () => {
    setup({ isAuthenticated: true, user: { id: '1', email: 'test@example.com' } });
    fireEvent.click(screen.getByRole('button', { name: /Log Out/i }));
    expect(mockLogout).toHaveBeenCalledTimes(1);
  });

  it('shows loading state', () => {
    setup({ isLoading: true });
    expect(screen.getByText(/Loading.../i)).toBeInTheDocument();
    expect(screen.getByRole('button', { name: /Log In \(Test\)/i })).toBeDisabled();
  });

  it('shows error message', () => {
    const testError = { message: 'Login failed' };
    setup({ error: testError });
    expect(screen.getByTestId('error-info')).toHaveTextContent('Login failed');
    // Match the actual call signature where a second undefined argument might be passed by React/Testing Library
    expect(ErrorInfo).toHaveBeenCalledWith(
      expect.objectContaining({ error: testError }),
      undefined
    );
  });

  describe('Compact mode', () => {
    it('renders compact not authenticated state', () => {
      setup({ compact: true });
      expect(screen.getByText(/Status:/i)).toHaveTextContent('Not Authenticated');
      expect(screen.getByRole('button', { name: /Login/i })).toBeEnabled();
      expect(screen.getByRole('button', { name: /Logout/i })).toBeDisabled();
      expect(screen.queryByText(/\(Debug\)/i)).not.toBeInTheDocument();
    });

    it('renders compact authenticated state with user email', () => {
      setup({
        isAuthenticated: true,
        user: { id: '1', email: 'compact@example.com' },
        compact: true,
      });
      expect(screen.getByText(/Status:/i)).toHaveTextContent('Authenticated');
      expect(screen.getByText(/compact@example.com/i)).toBeInTheDocument();
      expect(screen.queryByText(/User ID:/i)).not.toBeInTheDocument();
      expect(screen.getByRole('button', { name: /Login/i })).toBeDisabled();
      expect(screen.getByRole('button', { name: /Logout/i })).toBeEnabled();
    });

    it('calls login on compact login button click', async () => {
      setup({ compact: true });
      fireEvent.click(screen.getByRole('button', { name: /Login/i }));
      await waitFor(() => expect(mockLogin).toHaveBeenCalledTimes(1));
      expect(mockLogin).toHaveBeenCalledWith({
        email: 'user@example.com',
        password: 'password123',
      });
    });

    it('calls logout on compact logout button click', () => {
      setup({
        isAuthenticated: true,
        user: { id: '1', email: 'compact@example.com' },
        compact: true,
      });
      fireEvent.click(screen.getByRole('button', { name: /Logout/i }));
      expect(mockLogout).toHaveBeenCalledTimes(1);
    });
  });

  it('displays username if available in non-compact mode', () => {
    setup({
      isAuthenticated: true,
      user: { id: '1', email: 'test@example.com', username: 'TestUser' },
    });
    expect(screen.getByText(/Username: TestUser/i)).toBeInTheDocument();
  });

  it('does not display username if not available in non-compact mode', () => {
    setup({ isAuthenticated: true, user: { id: '1', email: 'test@example.com' } });
    expect(screen.queryByText(/Username:/i)).not.toBeInTheDocument();
    expect(screen.getByText(/Status:/i)).toHaveTextContent('Authenticated');
    expect(screen.getByText(/User ID: 1/i)).toBeInTheDocument();
    expect(screen.getByText(/Email: test@example.com/i)).toBeInTheDocument();
    expect(screen.getByRole('button', { name: /Log In \(Test\)/i })).toBeDisabled();
    expect(screen.getByRole('button', { name: /Log Out/i })).toBeEnabled();
  });

  it('does not display username in compact mode even if available', () => {
    setup({
      isAuthenticated: true,
      user: { id: '1', email: 'test@example.com', username: 'TestUser' },
      compact: true,
    });
    expect(screen.queryByText(/Username:/i)).not.toBeInTheDocument();
    expect(screen.getByText(/test@example.com/i)).toBeInTheDocument();
    expect(screen.getByText(/Status:/i)).toHaveTextContent('Authenticated');
    expect(screen.getByRole('button', { name: /Logout/i })).toBeEnabled();
  });

  it('matches snapshot in default state', () => {
    const { asFragment } = setup();
    expect(asFragment()).toMatchSnapshot();
  });

  it('matches snapshot in authenticated state', () => {
    const { asFragment } = setup({
      isAuthenticated: true,
      user: { id: '1', email: 'snap@example.com', username: 'SnapUser' },
    });
    expect(asFragment()).toMatchSnapshot();
  });

  it('matches snapshot in compact authenticated state', () => {
    const { asFragment } = setup({
      isAuthenticated: true,
      user: { id: '1', email: 'snap.compact@example.com' },
      compact: true,
    });
    expect(asFragment()).toMatchSnapshot();
  });

  it('matches snapshot in loading state', () => {
    const { asFragment } = setup({ isLoading: true });
    expect(asFragment()).toMatchSnapshot();
  });

  it('matches snapshot in error state', () => {
    const { asFragment } = setup({ error: { message: 'Snapshot error' } });
    expect(asFragment()).toMatchSnapshot();
  });
});
