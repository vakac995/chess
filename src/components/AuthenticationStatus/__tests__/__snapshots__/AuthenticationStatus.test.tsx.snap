// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AuthenticationStatus > matches snapshot in authenticated state 1`] = `
<DocumentFragment>
  <div
    class="my-4 rounded border bg-gray-100 shadow p-4"
  >
    <h2
      class="font-semibold mb-3 text-xl"
    >
      Authentication Status (Debug)
    </h2>
    <p
      class="mb-2"
    >
      Status: 
      <span
        class="font-medium text-green-600"
      >
        Authenticated
      </span>
    </p>
    <div
      class="mb-2 text-sm text-gray-700"
    >
      <p>
        User ID: 1
      </p>
      <p>
        Email: snap@example.com
      </p>
      <p>
        Username: SnapUser
      </p>
    </div>
    <div
      class="flex gap-2 mt-4"
    >
      <button
        class="rounded bg-blue-500 text-white transition duration-150 hover:bg-blue-600 disabled:cursor-not-allowed disabled:opacity-50 px-4 py-2"
        disabled=""
      >
        Log In (Test)
      </button>
      <button
        class="rounded bg-gray-500 text-white transition duration-150 hover:bg-gray-600 disabled:cursor-not-allowed disabled:opacity-50 px-4 py-2"
      >
        Log Out
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`AuthenticationStatus > matches snapshot in compact authenticated state 1`] = `
<DocumentFragment>
  <div
    class="my-4 rounded border bg-gray-100 shadow p-2"
  >
    <h2
      class="font-semibold mb-1 text-lg"
    >
      Authentication Status 
    </h2>
    <p
      class="mb-1 text-sm"
    >
      Status: 
      <span
        class="font-medium text-green-600"
      >
        Authenticated
      </span>
    </p>
    <div
      class="mb-1 text-xs text-gray-600"
    >
      <p>
        snap.compact@example.com
      </p>
    </div>
    <div
      class="flex gap-2 mt-2"
    >
      <button
        class="rounded bg-blue-500 text-white transition duration-150 hover:bg-blue-600 disabled:cursor-not-allowed disabled:opacity-50 px-2 py-1 text-sm"
        disabled=""
      >
        Login
      </button>
      <button
        class="rounded bg-gray-500 text-white transition duration-150 hover:bg-gray-600 disabled:cursor-not-allowed disabled:opacity-50 px-2 py-1 text-sm"
      >
        Logout
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`AuthenticationStatus > matches snapshot in default state 1`] = `
<DocumentFragment>
  <div
    class="my-4 rounded border bg-gray-100 shadow p-4"
  >
    <h2
      class="font-semibold mb-3 text-xl"
    >
      Authentication Status (Debug)
    </h2>
    <p
      class="mb-2"
    >
      Status: 
      <span
        class="font-medium text-red-600"
      >
        Not Authenticated
      </span>
    </p>
    <div
      class="flex gap-2 mt-4"
    >
      <button
        class="rounded bg-blue-500 text-white transition duration-150 hover:bg-blue-600 disabled:cursor-not-allowed disabled:opacity-50 px-4 py-2"
      >
        Log In (Test)
      </button>
      <button
        class="rounded bg-gray-500 text-white transition duration-150 hover:bg-gray-600 disabled:cursor-not-allowed disabled:opacity-50 px-4 py-2"
        disabled=""
      >
        Log Out
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`AuthenticationStatus > matches snapshot in error state 1`] = `
<DocumentFragment>
  <div
    class="my-4 rounded border bg-gray-100 shadow p-4"
  >
    <h2
      class="font-semibold mb-3 text-xl"
    >
      Authentication Status (Debug)
    </h2>
    <p
      class="mb-2"
    >
      Status: 
      <span
        class="font-medium text-red-600"
      >
        Not Authenticated
      </span>
    </p>
    <div
      class="rounded border border-red-300 bg-red-100 my-2 p-2"
    >
      <div
        data-testid="error-info"
      >
        Snapshot error
      </div>
    </div>
    <div
      class="flex gap-2 mt-4"
    >
      <button
        class="rounded bg-blue-500 text-white transition duration-150 hover:bg-blue-600 disabled:cursor-not-allowed disabled:opacity-50 px-4 py-2"
      >
        Log In (Test)
      </button>
      <button
        class="rounded bg-gray-500 text-white transition duration-150 hover:bg-gray-600 disabled:cursor-not-allowed disabled:opacity-50 px-4 py-2"
        disabled=""
      >
        Log Out
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`AuthenticationStatus > matches snapshot in loading state 1`] = `
<DocumentFragment>
  <div
    class="my-4 rounded border bg-gray-100 shadow p-4"
  >
    <h2
      class="font-semibold mb-3 text-xl"
    >
      Authentication Status (Debug)
    </h2>
    <p
      class="mb-2"
    >
      Status: 
      <span
        class="font-medium text-red-600"
      >
        Not Authenticated
      </span>
    </p>
    <p
      class="animate-pulse text-blue-500"
    >
      Loading...
    </p>
    <div
      class="flex gap-2 mt-4"
    >
      <button
        class="rounded bg-blue-500 text-white transition duration-150 hover:bg-blue-600 disabled:cursor-not-allowed disabled:opacity-50 px-4 py-2"
        disabled=""
      >
        Log In (Test)
      </button>
      <button
        class="rounded bg-gray-500 text-white transition duration-150 hover:bg-gray-600 disabled:cursor-not-allowed disabled:opacity-50 px-4 py-2"
        disabled=""
      >
        Log Out
      </button>
    </div>
  </div>
</DocumentFragment>
`;
